base/src/main/java/org/aya/concrete/stmt/TeleDecl.java

```diff
@@ -24,24 +24,23 @@
 import java.util.EnumSet;
 
 /**
- * Concrete definition, corresponding to {@link Def}.
+ * Concrete telescopic definition, corresponding to {@link Def}.
  *
  * @author re-xyr
+ * @see Decl
  */
-public sealed abstract class TopTeleDecl extends BaseDecl.Telescopic implements TopLevelDecl {
-  public final @NotNull Accessibility accessibility;
+public sealed abstract class TeleDecl extends CommonDecl implements Decl.Telescopic, Decl.TopLevel, Decl.Resulted {
+  public final @NotNull Decl.Personality personality;
   public @Nullable Context ctx = null;
   public @NotNull Expr result;
-  public final @NotNull TopLevelDecl.Personality personality;
+  // will change after resolve
+  public @NotNull ImmutableSeq<Expr.Param> telescope;
+  public @Nullable Def.Signature signature;
```

Comment:
Ditto

Reply:
These are used directly after pattern matching so it might be unnecessary to make them private. We can move to getters/setters in the future when a hook is needed.

Reply:
IC, thx!